federation-2/supergraph/src/command/compose.rs

```diff
@@ -13,7 +15,7 @@ pub struct Compose {
     ///
     /// NOTE: Each subgraph entry MUST contain raw SDL
     /// as the schema source.
-    config_file: Utf8PathBuf,
+    config_file: Option<Utf8PathBuf>,
```

Comment:
i would want to design this to require sending `-` explicitly to specify that input is coming from `stdin`. check out `FileDescriptorType` in rover: https://github.com/apollographql/rover/blob/main/src/utils/parsers.rs#L14

The `FromStr` is what lets you specify either `-` or a file: 

```rust
impl FromStr for FileDescriptorType {
    type Err = io::Error;

    fn from_str(input: &str) -> std::result::Result<Self, Self::Err> {
        if input == "-" {
            Ok(FileDescriptorType::Stdin)
        } else if input.is_empty() {
            Err(io::Error::new(
                io::ErrorKind::InvalidInput,
                anyhow!("The file path you specified is an empty string, which is invalid."),
            ))
        } else {
            let path = Utf8PathBuf::from(input);
            Ok(FileDescriptorType::File(path))
        }
    }
}
```